app-backend/tool/src/main/scala/params/EvalParams.scala

```diff
@@ -5,19 +5,57 @@ import java.util.UUID
 import cats.syntax.either._
 import com.azavea.rf.tool.ast._
 import io.circe._
+import io.circe.syntax._
+import io.circe.generic.JsonCodec
 
-case class EvalParams(sources: Map[UUID, RFMLRaster] = Map(), metadata: Map[UUID, NodeMetadata] = Map())
+/* Yes, it is correct that all three of these fields can be empty. It is
+ * possible to construct a legal AST that will produce results yet has no
+ * external data sources.
+ */
+case class EvalParams(
+  sources: Map[UUID, RFMLRaster] = Map(),
+  metadata: Map[UUID, NodeMetadata] = Map(),
+  overrides: Map[UUID, ParamOverride] = Map()
+)
 
 object EvalParams {
+
   implicit val encodeEvalParams: Encoder[EvalParams] =
-    Encoder.forProduct2("sources", "metadata")(ep => (ep.sources, ep.metadata))
+    Encoder.forProduct3("sources", "metadata", "overrides")(ep => (ep.sources, ep.metadata, ep.overrides))
 
   implicit val decodeEvalParams: Decoder[EvalParams] = new Decoder[EvalParams] {
     final def apply(c: HCursor): Decoder.Result[EvalParams] = {
-      val sources = c.get[Map[UUID, RFMLRaster]]("sources").getOrElse(Map())
-      val overrides = c.get[Map[UUID, NodeMetadata]]("metadata").getOrElse(Map())
+      val sources = c.get[Map[UUID, RFMLRaster]]("sources").getOrElse(Map.empty)
+      val metas = c.get[Map[UUID, NodeMetadata]]("metadata").getOrElse(Map.empty)
+      val overrides = c.get[Map[UUID, ParamOverride]]("overrides").getOrElse(Map.empty)
+
+      Right(EvalParams(sources, metas, overrides))
+    }
+  }
+}
+
+/** A sum type which allows us hold override values of varying types in the
+  * same `Map` in a type safe way.
+  */
+sealed trait ParamOverride
+
+object ParamOverride {
+
+  @JsonCodec case class Classification(classMap: ClassMap) extends ParamOverride
+  @JsonCodec case class Constant(constant: Double) extends ParamOverride
```

Comment:
Not sure if there's a rule here, but It seems to be stylistic preference to put annotations on the line prior to class definition